### DISCUSSION TITLE
Translate to more languages

### DISCUSSION THREAD

**Discussion by redphx** [redphx/better-xcloud#131] [Translations]
Help translating Better xCloud to more languages: https://crowdin.com/project/better-xcloud  
Let me know if the language you want to translate isn't in the list.  

Normally I don't want to publish a new version with new translation(s) only. I prefer having new features along with the translations. Thanks for understanding.

Thanks.

**Comment by redphx** [comment]
@ricardo404: there are a few more texts left, please help me translate it.

**Reply by ricardo404** [reply]
Sure, Red, I'll translate here, 1 minute.

**Reply by redphx** [reply]
Thanks. I can't release the new version until it's finised.

**Reply by ricardo404** [reply]
It's done, Red. Awaiting release.

**Reply by redphx** [reply]
1.17 released. Thanks for the help.

**Reply by redphx** [reply]
@ricardo404 there is one new string. Could you please translate it? Thanks.

**Reply by raphaelmnobrega** [reply]
Is the translation to pt-br all done?

**Comment by nyavana** [comment]
Chinese translation is complete, and please let me know if there is anything else you need.

**Reply by redphx** [reply]
Thanks for the help. You can test the translation with this version: https://github.com/redphx/better-xcloud/raw/main/better-xcloud.user.js

**Reply by nyavana** [reply]
I have made a few minor adjustments, and it should be good to go

**Reply by redphx** [reply]
Thanks a lot. I'll release a new version with your translation today.

**Comment by redphx** [comment]
Hi @rightones, I added your Korean translation, you can test it here: https://github.com/redphx/better-xcloud/raw/main/better-xcloud.user.js  
I'll publish a new version later today.  
Thanks.

**Comment by aleksishere** [comment]
Hi, I would like to translate Better xCloud to Polish. I'm asking to add a new language on Crowdin. 
Thanks!

**Reply by redphx** [reply]
Added: https://crowdin.com/translate/better-xcloud/14/en-pl  
Thanks

**Reply by redphx** [reply]
You can test your translation with this version: https://github.com/redphx/better-xcloud/raw/main/better-xcloud.user.js

**Reply by aleksishere** [reply]
I did some testing and looks all good to me.

**Reply by redphx** [reply]
Added in 1.18. Thanks for your help.

**Comment by PabloSebas** [comment]
Hello! I wanted to mention that I have already made the translation into Spanish 100%.

**Reply by redphx** [reply]
Thanks! You can test your translation with this version: https://github.com/redphx/better-xcloud/raw/main/better-xcloud.user.js

**Reply by PabloSebas** [reply]
In the testing I saw that there were two texts placed incorrectly, I already corrected the two text errors that I translated incorrectly, except for those two details mentioned that I already solved, everything would be perfect.

**Reply by redphx** [reply]
Updated: https://github.com/redphx/better-xcloud/raw/main/better-xcloud.user.js

**Reply by redphx** [reply]
Added in 1.18. Thanks for your help.

**Comment by KingNothing81** [comment]
Hi @redphx  ...

I've completed the German Translation yesterday. 
But I'm not satisfied yet with some of my translations. So I want to spend a little more time to improve things.

I will report back when I think it's good enough to be pushed out.

**Reply by KingNothing81** [reply]
@redphx: I think the German translation is ready for a test drive now.

**Reply by redphx** [reply]
Thanks for the help. You can test your translation with this: https://github.com/redphx/better-xcloud/raw/main/better-xcloud.user.js

**Reply by KingNothing81** [reply]
The translated string for “Custom Button Layout Style” looked too long. So I shortened it. I also found a shorter term for “Muted Colors”.

So you will have to approve those two strings again. 
Apart from that, I think it’s ok.

**Reply by KingNothing81** [reply]
Sorry ... shortened one more string for cosmetic reasons ("Show the estimated wait time"). So it also needs re-approval.
But now I'm done. I promise. =)

**Reply by redphx** [reply]
Added in 1.18. Thanks for your help.

**Comment by TheRedM** [comment]
I'll translate to Arabic, where to start?

**Reply by redphx** [reply]
Thank you, but Better xCloud doesn't support RTL languages at the moment. Adding support requires a lot of time, which I don't have at the moment. Thanks for understanding.

**Comment by glebanych** [comment]
Hey. I will gladly translate Better xCloud to Ukrainian :)

**Reply by redphx** [reply]
Added. Thanks.

**Reply by glebanych** [reply]
Hi. I just tested it using the link you provided above (https://github.com/redphx/better-xcloud/raw/main/better-xcloud.user.js). It's good, however I'm not able to find one string. There is a drop-down menu of languages in Better xCloud, and "Ukrainian" should be "українська", not "українець". I was not able to find that on Crowdin, so maybe it was auto-translated.
<img width="497" alt="Screenshot 2023-12-01 at 22 39 53" src="https://github.com/redphx/better-xcloud/assets/9075322/ac9ae1a9-f1b1-4ba3-8eef-e73c78e9e929">

**Reply by redphx** [reply]
My bad. I search that word myself. Will fix it soon. Thanks for the help.

**Reply by hectorevil** [reply]
Preferred game's language its  Idioma preferido del juego instead Idioma preferencial del juego

**Reply by redphx** [reply]
@hectorevil please create an issue on Crowdin

**Comment by Yunus2008** [comment]
Can you add Azerbaijani language to translate, i have many friends who use bxc and I want to translate it.

**Reply by redphx** [reply]
Is it left-to-right or right-to-left language? If it's RTL then Better xCloud won't be able to support it. Sorry.

**Comment by pawattoto** [comment]
Hello, I'm happy to translate Better xCloud into Thai for you if you need it.

**Reply by redphx** [reply]
Added: https://crowdin.com/project/better-xcloud/th  
Thanks

**Comment by Marxally** [comment]
Hello. I'd like to translate BXC to Catalan, if possible. Could you please enable the translation on Crowdin? Cheers!

**Reply by redphx** [reply]
Added: https://crowdin.com/project/better-xcloud/ca  
Please also translate the word "Catalan" in your own langue here (so I can add it to the language list)

**Reply by Marxally** [reply]
Thanks! It should be Català

**Reply by redphx** [reply]
You can test your translations with this version: https://github.com/redphx/better-xcloud/raw/typescript/dist/better-xcloud.user.js  

If you use Android app, then App settings > Better xCloud script version > Beta > Update script

**Reply by redphx** [reply]
Also thanks for the donation!

**Reply by Marxally** [reply]
> You can test your translations with this version: https://github.com/redphx/better-xcloud/raw/typescript/dist/better-xcloud.user.js
> 
> If you use Android app, then App settings > Better xCloud script version > Beta > Update script

Perfect! Thank you :)

**Comment by KingNothing81** [comment]
Hi @redphx …

Because I wanted to check the length of my translation of the string **“Touch control layout by %{name}“**, I noticed that the toast message shows the Indonesian version, not my expected German translation.

This happens with the current stable and beta version of the script. Although I haven’t seen other toast messages showing up with the wrong language so far, I don’t know if this might be a more general problem. So maybe you can have a look at this.

Thanks.

Edit: Just an observation, but I think this might be related to adding Catalan. Because now “Bahasa Indonesia” is on position 3 in the language selection, where “Deutsch” (German) was before. If this is related, this might affect all languages.

![PNG-Bild](https://github.com/redphx/better-xcloud/assets/146887651/cd1ebff2-f7b1-4fe1-8370-65ecf49da0d3)

**Reply by KingNothing81** [reply]
I think I found the problem ... in the script the supported languages are listed like this:

> var SUPPORTED_LANGUAGES = {
>   "en-US": "English (United States)",
>   "ca-CA": "Català",
>   "en-ID": "Bahasa Indonesia",
>   "de-DE": "Deutsch",
> (....)
> 

So "Indonesian" is on position 3, "German" on 4. 

But in the script's passages that contain the translated strings, it's the other way around: 
"German" is on position 3, "Indonesian" on 4. 

For example here in the mentioned string:

> "touch-control-layout": "Touch control layout",
>   "touch-control-layout-by": [
>     (e) => `Touch control layout by ${e.name}`,
>     (e) => `Format del control tàctil per ${e.name}`,
>     (e) => `Touch-Steuerungslayout von ${e.name}`, <- German
>     (e) => `Tata letak Sentuhan layar oleh ${e.name}`, <- Indonesian
> (...)

So somehow "Indonesia" and "German" got swapped. But as far as I can see, the order of the remaining languages is correct. So there should be no problem with those.

**Comment by Zakof1983** [comment]
Hi there, I would like to translate Better xCloud to Traditional Chinese. 
Thanks!

**Reply by redphx** [reply]
Added: https://crowdin.com/project/better-xcloud/zh-TW  
Thanks

**Reply by Zakof1983** [reply]
Hello, my translation has been completed, if there is anything else I need to submit or note, please let me know again, thank you!

**Reply by redphx** [reply]
That was fast. Thanks for the contribution. I'll add it in the next version.

**Comment by RKoldborg** [comment]
Hi there. Please add Danish to the list of languages. I'd gladly start translating 😉

**Reply by redphx** [reply]
Added: https://crowdin.com/editor/better-xcloud/14/en-da?view=comfortable  
Please also tell me how to write "Danish" in Danish too.  
Thanks.

**Reply by RKoldborg** [reply]
Wow, very fast! Danish in Danish is dansk 😊

**Reply by RKoldborg** [reply]
@redphx initial translation to Danish is done 😊

**Reply by redphx** [reply]
Thanks. I'll include it in the next version.

**Comment by Martincz05** [comment]
Hello,
Could you please added Czech language? **(Čeština)** I want to translate it.

**Reply by redphx** [reply]
Added https://crowdin.com/project/better-xcloud/cs  
Thanks

**Reply by Martincz05** [reply]
It should be done.

**Reply by redphx** [reply]
Thank you. I'll add it in the next release.
